portable/GCC/ARM_CM4F/portmacro.h

```diff
@@ -51,7 +51,7 @@
     #define portLONG          long
     #define portSHORT         short
     #define portSTACK_TYPE    uint32_t
-    #define portBASE_TYPE     long
+    #define portBASE_TYPE     uint32_t
```

Comment:
I don't think it's ok to change this to an unsigned type - it is meant to be signed as per every other port and as it has always been.  Changing it will probably cause pain for many users.  If it is changed, which I advise strongly against, it will also need to be changed in all the ARMv7-M and ARMv8-M ports.

Reply:
portBASE_TYPE is actually a deprecated type, kept only so application code still builds.  Recent code should be using BaseType_t.

Reply:
Thank you for your review and information. Currently, It caused the compile error because the warnings are treated as errors. What is your suggestion to fix this?

Reply:
Do you know which [specific warning flag](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/CMakeLists.txt#L251-L263) is giving you an error? The pedantic flag is something we've discussed removing in other repos in the past. If this is causing your issue - we might want to consider removing it.

Reply:
Also, which line is causing the error? Is it the kernel code or the application code?

Reply:
Sorry, this was -Wconversion flag in my application code.